enclave-runner/src/usercalls/mod.rs

```diff
@@ -681,21 +742,49 @@ impl RunningTcs {
         local_addr: Option<&mut OutputBuffer>,
         peer_addr: Option<&mut OutputBuffer>,
     ) -> IoResult<Fd> {
-        let addr = str::from_utf8(addr).map_err(|_| IoErrorKind::ConnectionRefused)?;
-        let stream = TcpStream::connect(addr)?;
-        if let Some(local_addr) = local_addr {
-            match stream.local_addr() {
-                Ok(local) => local_addr.set(local.to_string().into_bytes()),
-                Err(_) => local_addr.set(&b"error"[..]),
+        let mut l = String::new();
+        let mut p = String::new();
+        let stream_ext = 
+        {
+            let la = local_addr.as_ref().map(|_| &mut l);
+            let pa = peer_addr.as_ref().map(|_| &mut p);
+
+            self.enclave.usercall_ext.connect_stream(addr, la, pa)?
+        };
```

Comment:
Avoid very short names (i.e. 1 or 2 character names) `l`, `p`, `la`, `pa`.

If the stream returns `None`, we can/should ignore what `usercall_ext.connect_stream()` pushed to `local_addr` and `peer_addr`, since `None` indicates that the extension doesn't want to handle the usercall.

It might be a little clearer to write:
```
{
    let mut local_addr_str = local_addr.as_ref().map(|_| String::new());
    let mut peer_addr_str = peer_addr.as_ref().map(|_| String::new());
    if let Some(stream_ext) = self.enclave.usercall_ext
            .connect_stream(addr, local_addr_str.as_mut(), peer_addr_str.as_mut())? {
        if let Some(local_addr_str) = local_addr_str {
            local_addr.set(local_addr_str.into_bytes());
        }
        if let Some(peer_addr_str) = peer_addr_str {
            peer_addr.set(peer_addr_string.into_bytes());
        }
        return Ok(self.alloc_fd(FileDesc::Stream(stream)));
    }
}
// original function body before this PR would go here
```

Reply:
Thanks. I had lot of variables with short names to avoid borrow-checker issues. But I guess this is the cleanest way to achieve the same.

Reply:
I have added this to current push. Changed a bit as needed for making addr as str and other minor compiler fixes.